tests/beta/tapes/test_tape.py

```diff
@@ -994,3 +994,533 @@ def test_multiple_output_values(self, tol):
 
         res = tape.execute(dev)
         assert res.shape == (2,)
+
+
+class TestGradMethod:
+    """Tests for parameter gradient methods"""
+
+    def test_non_differentiable(self):
+        """Test that a non-differentiable parameter is
+        correctly marked"""
+        psi = np.array([1, 0, 1, 0]) / np.sqrt(2)
+
+        with QuantumTape() as tape:
+            qml.QubitStateVector(psi, wires=[0, 1])
+            qml.RX(0.543, wires=[0])
+            qml.RY(-0.654, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            probs(wires=[0, 1])
+
+        assert tape._grad_method(0) is None
```

Comment:
Are state prep parameters automatically nondiffable?

Reply:
No, state preps _can_ be differentiable. But in this case, `QubitStateVector` specifically isn't